src/backend/drm/surface/gbm.rs

```diff
@@ -445,6 +451,33 @@ impl Drop for FbHandle {
     }
 }
 
+struct BufferWrapper<'a, B>(&'a B);
```

Comment:
We could try to promote the already `pub` `framebuffer_from_bo` function from `drm::compositor::gbm` to some drm util module and re-use the logic here. I missed that opportunity in the planes PR unfortunately.

Reply:
Done in https://github.com/Smithay/smithay/pull/935/commits/6d32c24f8a4cc2cd6047d817d2196f9aea35e026